core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -224,20 +227,30 @@ public ClientFactoryBuilder connectionPoolListener(
     }
 
     /**
+     * Sets the {@link MeterRegistry} which collects various stats.
+     */
+    public ClientFactoryBuilder meterRegistry(MeterRegistry meterRegistry) {
+        this.meterRegistry = requireNonNull(meterRegistry, "meterRegistry");
+        return this;
+    }
+
+    /**
      * Returns a newly-created {@link ClientFactory} based on the properties of this builder.
      */
     public ClientFactory build() {
+        final MeterRegistry meterRegistry = this.meterRegistry != null ? this.meterRegistry
```

Comment:
What's the benefit of doing so? `firstNonNull()` will pre-instantiate the fallback registry instance unnecessarily and using `Optional` just defers the instantiation to `ServerConfig`.